src/odb/include/odb/db.h

```diff
@@ -971,6 +983,27 @@ class dbBlock : public dbObject
   ///
   void addBTermGroup(const std::vector<dbBTerm*>& bterms, bool order);
 
+  ///
+  /// Define the top layer grid for pin placement.
+  ///
+  void setBTermTopLayerGrid(dbTechLayer* layer,
```

Comment:
Why doesn't this take a `const dbBTermTopLayerGrid&`?  The fields are optional in the struct but mandatory in this API.

Reply:
I'm using this function in the odb.tcl to define the top layer grid. I understand that if I set the parameter as a `const dbBTermTopLayerGrid&`, it would require some swig work so Tcl can understand it.

Reply:
You could pass the values from tcl to the .i discretely and assemble the structure in the .i file if needed.